torch/csrc/autograd/python_variable_indexing.cpp

```diff
@@ -383,16 +306,30 @@ int THPVariable_setitem(PyObject* self, PyObject* index, PyObject* py_value) {
     // real 0-sized shapes.
     return 0;
   } else if (index == Py_Ellipsis) {
-    copy_to(self_, value);
+    at::indexing::copy_to(self_, value);
     return 0;
   } else if (index == Py_None || index == Py_True) { // NOLINT(cppcoreguidelines-pro-type-cstyle-cast)
-    copy_to(self_.unsqueeze(0), value);
+    at::indexing::copy_to(at::indexing::handleNoneSingleDim(self_), value);
     return 0;
   } else if (THPUtils_checkLong(index)) {
-    copy_to(applySelect(self_, 0, index), value);
+    at::indexing::copy_to(at::indexing::handleIntegerSingleDim(
+      self_,
+      THPUtils_unpackLong(index),
+      THPVariable_Check(index) ? THPVariable_Unpack(index) : at::indexing::undefined_tensor), value);
```

Comment:
This ternary seems very suspicious. How can an `index` be both a Long and a tensor?